Extensions/PlatformBehavior/platformerobjectruntimebehavior.ts

```diff
@@ -1679,6 +1905,205 @@ namespace gdjs {
+          ownerMinX = Math.min(ownerMinX, vertex[0]);
+          ownerMaxX = Math.max(ownerMaxX, vertex[0]);
+          ownerMinY = Math.min(ownerMinY, vertex[1]);
+          ownerMaxY = Math.max(ownerMaxY, vertex[1]);
+        }
+      }
+
+      this.ownerMinX = ownerMinX;
+      this.ownerMaxX = ownerMaxX;
+      this.headMinY = ownerMinY + upwardDeltaY;
+      this.ownerMinY = ownerMinY;
+      this.headMaxY = ownerMinY + downwardDeltaY;
+      this.floorMinY = ownerMaxY + upwardDeltaY;
+      this.ownerMaxY = ownerMaxY;
+      this.floorMaxY = ownerMaxY + downwardDeltaY;
+
+      this.allowedMinDeltaY = upwardDeltaY;
+      // Number.MAX_VALUE and not downwardDeltaY
+      // because it would means that a platform was found.
+      // see isCollidingAnyPlatform()
+      this.allowedMaxDeltaY = Number.MAX_VALUE;
+    }
+
+    initializeBeforeHitboxCheck() {
+      this.foundOverHead = false;
+      this.foundUnderHead = false;
+    }
+
+    /**
+     * Revert the search variables to a given state.
+     *
+     * This is used to revert side effect of jumpthru check.
+     * @param previousAllowedMinDeltaY
+     * @param previousAllowedMaxDeltaY
+     */
+    revertTo(previousAllowedMinDeltaY: float, previousAllowedMaxDeltaY: float) {
+      // Other members are either constants or local to an hitbox search.
+      this.allowedMinDeltaY = previousAllowedMinDeltaY;
+      this.allowedMaxDeltaY = previousAllowedMaxDeltaY;
+    }
+
+    setFloorIsTooHigh() {
+      this.allowedMinDeltaY = Number.MAX_VALUE;
+      this.allowedMaxDeltaY = -Number.MAX_VALUE;
+    }
+
+    floorIsTooHigh(): boolean {
+      // Return true when the 2 constraints are incompatible.
+      return this.allowedMinDeltaY > this.allowedMaxDeltaY;
+    }
+
+    isCollidingAnyPlatform(): boolean {
+      return this.ownerMaxY + this.allowedMaxDeltaY <= this.floorMaxY;
+    }
+
+    getFloorDeltaY(): float {
+      return this.allowedMaxDeltaY;
+    }
+
+    /**
+     * Check if the character can follow a given Y or move not to touch it
+     * and update the context with this new constraint.
+     * @param y
+     */
+    addPointConstraint(y: float): void {
+      if (y < this.floorMinY) {
+        // The platform is too high to walk on...
+        if (y > this.headMaxY) {
+          // ...but not over the object.
+          this.setFloorIsTooHigh();
+          return;
+        }
+        // ...but over the object.
+        this.foundOverHead = true;
+        if (this.foundUnderHead) {
+          this.setFloorIsTooHigh();
+          return;
+        }
+        // When there is a platform on the top,
+        // the character is constraint on how high
+        // he can follow a floor.
+        this.allowedMinDeltaY = Math.max(
+          this.allowedMinDeltaY,
+          y - this.ownerMinY
+        );
+      } else {
+        // The platform can be walked on.
+        this.foundUnderHead = true;
+        if (this.foundOverHead) {
+          this.setFloorIsTooHigh();
```

Comment:
Can you add a comment for this line. If I understand correctly:
- we found a point under the top of the character.
- we found a point over the top of the character.

Could you clarify in a comment why we're sure that this means the floor is too high? Thanks!